test/vector_add.cpp

```diff
@@ -0,0 +1,343 @@
+    auto idx          = miopen::mismatch_idx(t_gpu1.data, t_gpu2.data, compare_equal<T>);
+    bool valid_result = idx >= miopen::range_distance(t_gpu2);
+
+    if(!valid_result)
+    {
+        std::cout << "diff at:" << idx << ", gpu1:" << t_gpu1[idx] << ", gpu2:" << t_gpu2[idx]
+                  << std::endl;
+    }
+    return valid_result;
+}
+
+// Generate numbers between 0 and 100
+template <typename T>
+void rand_tensor(tensor<T>& t, int max = 100, int min = 0) 
+{
+    // use integer to random.
+    for(size_t i = 0; i < t.data.size(); i++)
+        t[i] = static_cast<T>(prng::gen_A_to_B(min, max));
+}
+
+// Compare the three tensors
+template <typename T>
+bool verify_tensor_cpu_ocl_hip(tensor<T>& t_cpu, tensor<T>& t_gpu_ocl, tensor<T>& t_gpu_hip)
+{
+    EXPECT(t_gpu_ocl.data.size() == t_cpu.data.size() && t_gpu_hip.data.size() == t_cpu.data.size()); // Make sure that the tensors are of the same size
+
+    auto idx_hip_cpu      = miopen::mismatch_idx(t_gpu_hip.data, t_cpu.data, compare_equal<T>);
+
+    auto idx_ocl_hip      = miopen::mismatch_idx(t_gpu_ocl.data, t_gpu_hip.data, compare_equal<T>);
+
+    
+    bool r_hip_cpu = idx_hip_cpu >= miopen::range_distance(t_cpu);
+    bool r_ocl_hip = idx_ocl_hip >= miopen::range_distance(t_gpu_ocl);
+
+    if(!(r_hip_cpu && r_ocl_hip))
+    {
+        std::cout << "diff at:" << idx_hip_cpu << ", hip:" << t_gpu_hip[idx_hip_cpu] << ", cpu:" << t_cpu[idx_hip_cpu]
+                  << std::endl;
+        std::cout << "diff at:" << idx_ocl_hip << ", ocl:" << t_gpu_ocl[idx_ocl_hip] << ", hip:" << t_gpu_hip[idx_ocl_hip]
+                  << std::endl;
+    }
+    return (r_hip_cpu && r_ocl_hip);
+}
+
+
+template <class T>
+void vec_add(const tensor<T>& srcA,
+                  const tensor<T>& srcB,
+                  tensor<T>& dstC)
+{
+   
+    for(size_t i = 0; i < SIZE_TENSOR; i++)
+    {
+        dstC[i] = srcA[i] + srcB[i];
+    }
+
+}
+
+template <class T>
+struct verify_vecadd_ocl
+{
+    tensor<T> srcA;
+    tensor<T> srcB;
+    tensor<T> dstC;
+
+    verify_vecadd_ocl(const tensor<T>& p_srcA,
+                        const tensor<T>& p_srcB,
+                        const tensor<T>& p_dstC)
+    {
+        srcA      = p_srcA;
+        srcB      = p_srcB;
+        dstC      = p_dstC;
+
+    }
+
+    tensor<T> cpu() const
+    {
+        auto r = dstC;
+
+        vec_add(srcA, srcB, r);
+
+        return r;
+    }
+
+    tensor<T> gpu() const
+    {
+        auto r        = dstC;
+        auto&& handle = get_handle();
+        auto srcA_dev  = handle.Write(srcA.data);
+        auto srcB_dev  = handle.Write(srcB.data);
+
+        // clear the destination tensor
+        for(size_t i = 0; i < r.data.size(); i++)
+            r[i] = 0;
+
+        auto dstC_dev  = handle.Write(r.data);
+
+
+        std::string program_name = "MIOpenVecAddOCL.cl";
+        std::string kernel_name  = "vector_add";
+
+        std::string network_config = "NW_CONFIG_PLACEHOLDER";
```

Comment:
this should be a descriptive and somewhat unique name that identifies the operation. Something like `standalone_kernel_vector_add_cl` and then similar for hip. The string makes up the key for the kernel cache, so it needs to be unique.